adala/utils/llm.py

```diff
@@ -110,40 +125,28 @@ async def async_get_llm_response(
     # constrained generation branch - use `response_model` to constrain the LLM response
     try:
         instructor_response, completion = await async_instructor_client.chat.completions.create_with_completion(
-            model=model,
-            api_key=api_key,
             messages=messages,
-            max_tokens=max_tokens,
-            temperature=temperature,
             response_model=response_model,
-            timeout=timeout,
+            **inference_settings.dict(),
```

Comment:
especially if we're just unpacking this one level down, I don't love adding an abstraction for this, as much as it may reduce the number of params. Imo it's kind of lying to ourselves about simplifying things when really we're just hiding things

Reply:
not the hill I will die on though

Reply:
I don't want it to be a lot of abstraction, but we were passing the same params in 7 (!) places - get_llm_response x2, async_get_llm_response x2, parallel_get_llm_response, sync runtime, and async runtime - bc we're calling the same (pair of) litellm function(s) every time. And they did not have parity with each other just for reasons of forgetting to add/update things, which is the wrong reason imo. Is there a less hidey way to do this? Maybe just a dict? Or put it in the util fns but not in the runtimes?

Reply:
alternatively we could reduce the number of utility fns...

Reply:
yeah exactly, the real issue is how many layers we go through before the actual litellm call...

Reply:
planning to reduce that in a future pr fwiw